src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonPropertyInfoOfT.cs

```diff
@@ -235,7 +235,10 @@ internal override bool GetMemberAndWriteJson(object obj, ref WriteStack state, U
             T value = Get!(obj);
 
             if (Options.ReferenceHandlingStrategy == ReferenceHandlingStrategy.IgnoreCycles &&
-                !Converter.IsValueType && value != null &&
+                value != null &&
+                // .NET types that are parsed to JSON value types doesn't need to be tracked for cycle detection e.g: string.
+                // However JsonConverter<object> that uses ConverterStrategy == Value is an exception.
+                (Converter.CanBePolymorphic || (!Converter.IsValueType && ConverterStrategy != ConverterStrategy.Value)) &&
```

Comment:
Is the `CanBePolymorphic` check necessary? I believe polymorphic converters can neither be value types or have a `Value` converter strategy.

Reply:
`ObjectConverter` does have a `Value` converter strategy; that's why I added the `CanBePolymorphic` check, in order to also check for circularity in boxed types.

Reply:
Got it thanks. I think I was conflating `ObjectConverter` with `ObjectDefaultConverter` which are apparently different.